### PR TITLE
MCR-2398 MCRXMLMetadataManager Interface

### PR DESCRIPTION
**Pull request by OPNA2608**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2398).

~~Still need to test & find a good solution for code calling `MCRXMLMetadataManager.instance()` directly vs `MCRAbstractXMLMetadataManager.getManager()` reading the configured XML Metadata Manager class and instantiating it via its `instance()` method.~~

Fixed by moving MCRXMLMetadataManager -> MCRDefaultXMLMetadataManager & MCRAbstractXMLMetadataManager -> MCRXMLMetadataManager.

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on eb60ce75433cd952ac559146840f3e406434b150! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7087-pull_request-1234

**Comment by OPNA2608** [comment]
Gonna look over code comments again, otherwise ready for review.

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 1c68194d23490cea10dc3477f1e83b96b63841fc! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7122-pull_request-1234

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 1]
```diff
@@ -47,44 +32,44 @@
...
- *
- * The default IFS2 store is MCRVersioningMetadataStore, which
- * versions metadata using SVN in local repositories below SVNBase.
- * If you do not want versioning and would like to have better
- * performance, change the following property to
- *
- * MCR.Metadata.Store.DefaultClass=org.mycore.datamodel.ifs2.MCRMetadataStore
+ * 
+ * If an SVN-based store is configured, then the following property will be used to
+ * store & manage local SVN repositories:
```
Comment (rsteph-de) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: error: bad HTML entity:
* store & manage local SVN repositories:

Comment (OPNA2608) [reply mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: Fixed, thanks.

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 5a8bd8504a8e669b7e56322b7f1b977cf64677ef! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7140-pull_request-1234

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on fd38519db19d3a9dc595fce5d248e858e3ae0e68! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7144-pull_request-1234

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d0d8462ec22545b23fe39fa074bc7ddfb18d3d7f! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/7146-pull_request-1234

**Review by yagee-de — APPROVED** [review]
Looks good as the next step for me. But I think one refactoring would be useful:

- move all abstract methods to an interface and let MCRDefaultXMLMetadataManager inherit this interface and remove the superclass from MCRDefaultXMLMetadataManager
- make abstract methods public and delegate to the interface methods

Interfaces are a better preparation for java modules

**Comment by OPNA2608** [comment]
I've dropped all commits that were unrelated to the ticket's goal. I'll move them into a follow-up PR once this is merged.

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 2]
```diff
@@ -18,379 +18,289 @@
...
+     * If MCR.Metadata.Manager.Class is not set, an instance of {@link MCRDefaultXMLMetadataManager} is returned.
+     *
+     * @return an instance of the configured xml metadata manager if any is set, or MCRDefaultXMLMetadataManager
+     */
+    public static synchronized MCRXMLMetadataManager instance() {
+        if (SINGLETON == null) {
+            SINGLETON = new MCRXMLMetadataManager();
+        }
+        if (IMPLEMENTATION == null) {
+            IMPLEMENTATION = MCRConfiguration2
```
Comment (sebhofmann) [inline mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java:70]: The Property MCR.Metadata.Manager.Class should be defined in the mycore.properties.
We have the Rule that every default property should be in the mycore.properties, except for dynamic named properties.

**Review by sebhofmann — CHANGES_REQUESTED** [review]
Everything else looks good for me.

**Review by sebhofmann — APPROVED** [review]

**Pull request merged**